### PR TITLE
Source validation fixes

### PR DESCRIPTION
**Pull request by gabo1208**
Little bug fix to knative/eventing#5638

### CODE REVIEW

File: apis/duck/v1/source_types_test.go [hunk 1]
```diff
@@ -32,6 +32,10 @@ func TestSourceValidate(t *testing.T) {
 		name: "empty source validation",
 		src:  nil,
 		want: nil,
+	}, {
+		name: "empty source spec validation",
+		src:  &Source{Spec: SourceSpec{}},
```
Comment (n3wscott) [inline apis/duck/v1/source_types_test.go:37]: Can you also add a test where `.spec` is nil?

**Comment by mattmoor** [comment]
I don't see this cherry-picked into `release-0.26`, and it is causing https://github.com/knative/eventing/pull/5744 to fail

**Pull request merged**